libsolidity/analysis/TypeChecker.cpp

```diff
@@ -3845,6 +3846,17 @@ void TypeChecker::endVisit(Literal const& _literal)
 				)
 					// TODO: Is this triggered when the argument is out of range? Test.
 					parameterTypeMessage = "The type of the literal cannot be converted to the parameters of the suffix function.";
+
+				if (
+					auto exponentType = dynamic_cast<IntegerType const*>(suffixFunctionType->parameterTypes().at(1));
+					exponentType && exponentType->isSigned()
+				)
```

Comment:
This should go into the `else` branch above. Otherwise you can't assume that `suffixFunctionType` is not null.

Reply:
Sorry, I didn't understand this clearly. I assumed `suffixFunctionType` can't be null at this point because of a previous test at line `3802`.

Reply:
That's only if it's a fatal error. This one is not.

Reply:
Ah, understood it now. Thanks.

Reply:
But isn't this inside the `else` branch of that test? I mean, if `suffixFunctionType` is null, then this block doesn't execute.
Maybe I'm missing something...

Reply:
Damn, I think I got bamboozled by github's crappy rendering of tabs. For me it sometimes chooses with randomly for every nesting level and also, the width for some reason slightly shifts at transition between commentable and non-commentable code :|

Yeah, it seems fine now but I had to look closely and count indents to notice that...

Reply:
Yeah, sometimes the rendering of tabs mess with indents...it is really confusing.